docs/developers.rst

```diff
@@ -2,206 +2,214 @@
+
+::
+
+  black <py-file-you-created>
+
+it is usually best to then inspect the file and correct any strange choices `black` has made.
+
+Unit Tests
+==========
+
+Pull requests will require existing unit tests to pass before they can be merged. Additionally, new unit tests should be written for all new public methods and functions. Unit tests for each Likelihood and Theory should be placed in the tests directory with a name matching that of the python file in which the class is defined::
+
+ SOLikeT/soliket/tests/test_my_module.py
+
+
+For Likelihoods we request that there is a test which compares the result of a likelihood calculation to a precomputed expected value which is hard coded in the tests file, to a tolerance of ``1.e-3``::
+
+  assert np.isclose(loglike_just_computed, -25.053, rtol=1.e-3)
+
+For more advice on how to write tests see the `Astropy Testing Guidelines <https://docs.astropy.org/en/stable/development/testguide.html>`_.
+
+Tests run a set of SOLikeT calculations with known expected results. There are (at least) two reasons you might want to run tests:
+
+Checking code in development
+----------------------------
+To see if codes you have written when developing SOLikeT are valid and will pass the Continuous Integration (CI) tests which we require for merging on github.
+
+If you are using conda, the easiest way to run tests (and the way we run them) is to use tox-conda::
+
+  pip install tox-conda
+  tox -e test
+
+This will create a fresh virtual environment replicating the one which is used for CI then run the tests (i.e. without touching your current environment). Note that any args after a '--' string will be passed to pytest, so::
+
+  tox -e test -- -k my_new_module
+
+will only run tests which have names containing the string 'my_new_model', and ::
+
+  tox -e test -- -pdb
+
+will start a pdb debug instance when (sorry, *if*) a test fails.
+
+Checking environment configuration
+----------------------------------
+Check SOLikeT is working as intended in a python environment of your own specification (i.e. you have installed SOLikeT not using the soliket-tests conda environment).
+
+For this you need to make sure all of the required system-level and python dependencies described in `the installation instructions <install.html>`_ are working correctly, then run::
+
+  pytest -v soliket
+
+Good luck!
+
+If your unit tests check the statistical distribution of a random sample, the test outcome itself is a random variable, and the test will fail from time to time. Please mark such tests with the ``@pytest.mark.flaky`` decorator, so that they will be automatically tried again on failure. To prevent non-random test failures from being run multiple times, please isolate random statistical tests and deterministic tests in their own test cases.
```

Comment:
The last discussion on unit tests seems too detailed/technical. Should we put this in parenthesis - I'm worried that a significant number of people will be confused by it